### PR TITLE
fix(web): version and deployment patch editing

### PR DESCRIPTION
**Pull request by m8vago**

### CODE REVIEW

File: web/crux/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { Config } from '@jest/types';
+
+const config: Config.InitialOptions = {
+  moduleFileExtensions: ['js', 'json', 'ts'],
```
Comment (polaroi8d) [inline web/crux/jest.config.ts:4]: We need the `'js', 'json',` version?

File: web/crux/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  transform: {
+    '^.+\\.(t|j)s$': 'ts-jest',
+  },
+  collectCoverageFrom: ['src/server/**/*.(t|j)s'],
+  coveragePathIgnorePatterns: ['src/server/console', 'src/server/migration'],
+  coverageDirectory: 'coverage',
+  testEnvironment: 'node',
+};
+
+export default config;
```
Comment (polaroi8d) [inline web/crux/jest.config.ts:16]: Add newline please 🦅

File: protobuf/proto/crux.proto [hunk 3]
```diff
@@ -614,24 +620,36 @@ message Volume {
...
+message UniqueSecretKeyList { repeated UniqueSecretKey data = 1000; }
+
 message UniqueSecretKeyValue {
   string id = 100;
   string key = 101;
   string value = 103;
-  string publicKey = 104;
-  bool required = 105;
-  optional bool encrypted = 106;
+  bool required = 104;
```
Comment (polaroi8d) [inline protobuf/proto/crux.proto:647]: Why you changed the order?

File: web/crux/src/app/audit/audit.service.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export default class AuditService {
     const identites = await this.kratos.getIdentitiesByIds(auditLog.map(it => it.userId))
 
     return {
-      data: auditLog.map(it => this.mapper.toGrpc(it, identites)),
```
Comment (polaroi8d) [inline web/crux/src/app/audit/audit.service.ts:29]: We are using both of the formats, talk about this in person and refactor.

File: web/crux/src/app/deploy/deploy.service.ts [hunk 5]
```diff
@@ -413,9 +431,9 @@ export default class DeployService {
           )
 
           return {
-            common: this.mapper.configToCommonConfig(mergedConfig),
-            crane: this.mapper.configToCraneConfig(mergedConfig),
-            dagent: this.mapper.configToDagentConfig(mergedConfig),
+            common: this.mapper.configToAgentCommonConfig(mergedConfig),
+            crane: this.mapper.configToAgentCraneConfig(mergedConfig),
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.service.ts:435]: `configToAgentDagentConfig` I see why but `configToDagentConfig` better, no?

File: web/crux/src/app/image/image.service.ts [hunk 6]
```diff
@@ -36,7 +36,7 @@ export default class ImageService {
     })
 
     return {
-      data: images.map(it => this.mapper.toGrpc(it)),
+      data: images.map(it => this.mapper.detailsToProto(it)),
```
Comment (polaroi8d) [inline web/crux/src/app/image/image.service.ts:39]: Why `details`?

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: web/crux/jest.config.ts (hunk 1 above)
Comment (m8vago) [reply web/crux/jest.config.ts:4]: not really, this was the example config. i remove them

File: protobuf/proto/crux.proto (hunk 3 above)
Comment (m8vago) [reply protobuf/proto/crux.proto:647]: so the optional fields would be at the end. i figure if it needs to be modified anyway, it's not a big deal to change the order as well.

File: web/crux/src/app/audit/audit.service.ts (hunk 4 above)
Comment (m8vago) [reply web/crux/src/app/audit/audit.service.ts:29]: i didn't realized that one, while renaming it. :smile: It's indeed sounds silly. However there is already a `configToDagentConfig`. I rename them.

File: web/crux/src/app/image/image.service.ts (hunk 6 above)
Comment (m8vago) [reply web/crux/src/app/image/image.service.ts:39]: oops, nice catch! ty

File: web/crux-ui/src/server/crux/mappers/image-mappers.ts [hunk 7]
```diff
@@ -28,33 +28,47 @@ import {
...
   ImageResponse,
-  InitContainer,
   LogConfig,
   Volume,
 } from '@app/models/grpc/protobuf/proto/crux'
 import { timestampToUTC } from '@app/utils'
 import { registryTypeProtoToDto } from './registry-mappers'
 
+export const objectHasProperties = (object: any): boolean => {
+  const values = Object.values(object)
```
Comment (robot9706) [inline web/crux-ui/src/server/crux/mappers/image-mappers.ts:43]: `Object.values(object).some(it => !!it)` or something like that?

**Review by robot9706 — APPROVED** [review]
Minor comments, tested seems to be working fine 🚀

**Review by polaroi8d — APPROVED** [review]
LGTM, minors were discussed in person.

**Pull request merged**